contracts/clearing/TradingApp.sol

```diff
@@ -108,17 +113,15 @@ contract TradingApp is IForceMoveApp {
 		ITradingTypes.Settlement memory settlement,
 		RecoveredVariablePart[] calldata proof
 	) internal pure {
-		if (proof.length == 1) {
-			VariablePart memory proof = proof[0].variablePart;
-			ITradingTypes.Order memory order = abi.decode(proof.appData, (ITradingTypes.Order));
-			return;
-		}
-
 		bytes32[] memory orderIDs = new bytes32[](proof.length);
 		for (uint256 i = 0; i < proof.length - 1; i++) {
 			VariablePart memory currProof = proof[i].variablePart;
 			VariablePart memory nextProof = proof[i + 1].variablePart;
 
+			require(
+				currProof.turnNum >= 2 && nextProof.turnNum >= 2,
+				'only prefund and postfund can have turnNum < 2'
+			);
 			require(currProof.turnNum + 1 == nextProof.turnNum, 'turns are not consecutive');
```

Comment:
We can safely move this check out of the loop to save gas. We just need to check that the first state in the proof has turnNum >= 2, and by having the check that turn numbers are consecutive, we can guarantee that all of them have turnNum >= 2